Source/WebKit/webpushd/MockPushServiceConnection.h

```diff
@@ -39,19 +39,25 @@ class MockPushServiceConnection final : public PushServiceConnection {
     void subscribe(const String& topic, const Vector<uint8_t>& vapidPublicKey, SubscribeHandler&&) override;
     void unsubscribe(const String& topic, const Vector<uint8_t>& vapidPublicKey, UnsubscribeHandler&&) override;
 
-    Vector<String> enabledTopics() override;
-    Vector<String> ignoredTopics() override;
-    Vector<String> opportunisticTopics() override;
-    Vector<String> nonWakingTopics() override;
+    Vector<String> enabledTopics() override { return m_enabledTopics; }
```

Comment:
Can this be const and return a `const Vector<String>`?

Reply:
The problem here is that `ApplePushServiceConnection` (which is another subclass in this class hierarchy) has to return `Vector<String>` since it doesn't cache the topics locally in a member variable. Instead it always constructs a `Vector<String>` on the fly by converting it from the `NSArray<NSString *>` given to it from APS.